File: video-api/src/main/java/com/fiapql/videoapi/dto/VideoResponse.java
package com.fiapql.videoapi.dto;

import com.fiapql.videoapi.entity.VideoStatus;

import java.time.LocalDateTime;
import java.util.UUID;

public record VideoResponse(
    UUID          id,
    String        originalFilename,
    VideoStatus   status,
    String        downloadUrl,       // preenchido apenas quando COMPLETED
    LocalDateTime createdAt,
    LocalDateTime updatedAt
) {}

File: video-api/src/main/kotlin/com/fiapql/videoapi/dto/VideoResponse.kt
package com.fiapql.videoapi.dto

import com.fiapql.videoapi.entity.VideoStatus
import java.time.LocalDateTime
import java.util.UUID

data class VideoResponse(
    val id: UUID?,
    val originalFilename: String,
    val status: VideoStatus,
    val downloadUrl: String?,       // preenchido apenas quando COMPLETED
    val createdAt: LocalDateTime?,
    val updatedAt: LocalDateTime?
)
